### ISSUE TITLE
Transform at the best iteration for LightGBMClassifier

### ISSUE THREAD

**Issue by dspetukhov** [Azure/mmlspark#1006]
I train mmlspark LightGBMClassifier with early stopping using the parameters .setEarlyStoppingRound() and .setValidationIndicatorCol()

Is it possible to specify best iteration with highest validation metric value for `transform()` method similarly to parameter num_iteration in `predict()` method in the native LightGBM or parameter setTreeLimit in xgboost4j? Is there a way to access best iteration value? Is there a way to access train and validation metrics during training process similarly to `summary()` method in xgboost4j?

Unfortunately I have not found a way to make something out of this.
It would be great if these features will be added.

**Comment by imatiach-msft** [comment]
fixed with PR #1024
